src/main/java/io/javalin/websocket/WsContext.kt

```diff
@@ -54,6 +56,10 @@ abstract class WsContext(val sessionId: String, @JvmField val session: Session)
     fun <T> attribute(key: String): T? = upgradeCtx.attribute(key)
     fun <T> attributeMap(): Map<String, T?> = upgradeCtx.attributeMap()
 
+    fun sessionAttribute(key: String, value: Any?) = httpSession.setAttribute(key, value)
```

Comment:
Warning.

the HttpSession belongs to the HTTP portion of the upgrade.
and is attached to the HttpServletRequest object.
Once the connection is upgraded to WebSocket, the HTTP portion is recycled, and the HttpSession is also recycled / cleaned out.

Grab the attributes you need during the upgrade, and give them to the WebSocket implementation, not the entire HttpSession.

Reply:
The safest way to think is to do all HTTP related pieces during the upgrade, because once you are WebSocket, you are no longer HTTP.

Reply:
That's is very good to know, thank you! I was wondering if I should include this, I really only needed the session attributes to see if the upgrade should be allowed. I will remove all three methods.

Reply:
Turns out that I need to give developers access in `onConnect`, so I'll have to copy the map.